### PR TITLE
Add test cases for YamlReadwriteSplittingRuleConfigurationSwapper

### LINKED ISSUE #25353: Add test cases for YamlReadwriteSplittingRuleConfigurationSwapper
Please reference `YamlEncryptRuleConfigurationSwapperTest`

### PR DESCRIPTION
**Pull request by tobehardest**
Fixes #25353 

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by sandynz** [comment]
Hi @tobehardest , there's check style error, could you update the code?

```
Error:  /home/runner/work/shardingsphere/shardingsphere/features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/
YamlReadwriteSplittingRuleConfigurationSwapperTest.java:32:17: Using the '.*' form of import should be avoided - java.util.*. [AvoidStarImport]

Error:  /home/runner/work/shardingsphere/shardingsphere/features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/
YamlReadwriteSplittingRuleConfigurationSwapperTest.java:48: Line is longer than 200 characters (found 233). [LineLength]
```

And you could import `src/resources/code-style-idea.xml` into your IDEA Code Style.

**Comment by tobehardest** [comment]
Updated.

File: features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    
+    @Test
+    void assertSwapToObject() {
+        ReadwriteSplittingRuleConfiguration actual = getSwapper().swapToObject(creatYamlReadwriteSplittingRuleConfiguration());
+        assertThat(actual.getDataSources().size(), is(1));
+        assertThat(actual.getLoadBalancers().size(), is(1));
+    }
+    
+    private YamlReadwriteSplittingRuleConfiguration creatYamlReadwriteSplittingRuleConfiguration() {
```
Comment (sandynz) [inline features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java:63]: `creat` could be `create`

File: features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                new ReadwriteSplittingDataSourceRuleConfiguration("readwrite", "write_ds", Arrays.asList("read_ds_0", "read_ds_1"), "random"));
+        Map<String, AlgorithmConfiguration> loadBalancers = Collections.singletonMap("myLoadBalancer", new AlgorithmConfiguration("RANDOM", new Properties()));
+        return new ReadwriteSplittingRuleConfiguration(dataSources, loadBalancers);
+    }
+    
+    @Test
+    void assertSwapToObject() {
+        ReadwriteSplittingRuleConfiguration actual = getSwapper().swapToObject(creatYamlReadwriteSplittingRuleConfiguration());
+        assertThat(actual.getDataSources().size(), is(1));
+        assertThat(actual.getLoadBalancers().size(), is(1));
```
Comment (sandynz) [inline features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java:60]: Could we assert part of the important content?

File: features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        result.getDataSources().put("t_readwrite", dataSourceRuleConfig);
+        YamlAlgorithmConfiguration algorithmConfig = new YamlAlgorithmConfiguration();
+        algorithmConfig.setType("RANDOM");
+        result.getLoadBalancers().put("random_loadbalancer", algorithmConfig);
+        return result;
+    }
+    
+    private YamlReadwriteSplittingRuleConfigurationSwapper getSwapper() {
+        ReadwriteSplittingRuleConfiguration ruleConfig = mock(ReadwriteSplittingRuleConfiguration.class);
+        return (YamlReadwriteSplittingRuleConfigurationSwapper) OrderedSPILoader.getServices(YamlRuleConfigurationSwapper.class, Collections.singleton(ruleConfig)).get(ruleConfig);
```
Comment (sandynz) [inline features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java:77]: Could we just `new YamlReadwriteSplittingRuleConfigurationSwapper()` to simplify code?

File: features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+public class YamlReadwriteSplittingRuleConfigurationSwapperTest {
+    
+    @Test
+    void assertSwapToYamlConfiguration() {
+        YamlReadwriteSplittingRuleConfiguration actual = getSwapper().swapToYamlConfiguration(creatReadwriteSplittingRuleConfiguration());
+        assertThat(actual.getDataSources().size(), is(1));
+        assertThat(actual.getLoadBalancers().size(), is(1));
```
Comment (sandynz) [inline features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java:46]: Could we assert part of the important content?

**Review by sandynz — CHANGES_REQUESTED** [review]

**Comment by tobehardest** [comment]
please review again.

File: features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java [hunk 5]
```diff
@@ -35,45 +34,61 @@
...
         assertThat(actual.getDataSources().size(), is(1));
         assertThat(actual.getLoadBalancers().size(), is(1));
+        assertReadwriteSplittingRule(actual);
+    }
+
+    void assertReadwriteSplittingRule(final YamlReadwriteSplittingRuleConfiguration actual) {
+        assertNotNull(actual.getDataSources().get("readwrite"));
+        YamlReadwriteSplittingDataSourceRuleConfiguration config = actual.getDataSources().get("readwrite");
+        assertThat(config.getWriteDataSourceName(), CoreMatchers.is("write_ds"));
+        assertThat(actual.getDataSources().get("readwrite").getLoadBalancerName(), CoreMatchers.is("random"));
```
Comment (sandynz) [inline features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java:53]: `CoreMatchers.is` could be `is`

File: features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java [hunk 6]
```diff
@@ -35,45 +34,61 @@
...
+        dataSourceRuleConfig.setLoadBalancerName("random");
+        YamlReadwriteSplittingRuleConfiguration result = new YamlReadwriteSplittingRuleConfiguration();
         result.getDataSources().put("t_readwrite", dataSourceRuleConfig);
         YamlAlgorithmConfiguration algorithmConfig = new YamlAlgorithmConfiguration();
         algorithmConfig.setType("RANDOM");
         result.getLoadBalancers().put("random_loadbalancer", algorithmConfig);
         return result;
     }
-    
+
```
Comment (sandynz) [inline features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java]: Indentation in empty line between methods could be kept

**Review by sandynz — CHANGES_REQUESTED** [review]

**Comment by tobehardest** [comment]
OK.

**Review by sandynz — APPROVED** [review]
LGTM

**Comment by sandynz** [comment]
@tobehardest Thanks for your contribution

**Pull request merged**